lib/optimizely.rb

```diff
@@ -98,6 +105,16 @@ def initialize( # rubocop:disable Metrics/ParameterLists
 
       @notification_center = notification_center.is_a?(Optimizely::NotificationCenter) ? notification_center : NotificationCenter.new(@logger, @error_handler)
 
+      setup_odp!
```

Comment:
We're already utilizing the notification center callback to update the config- I wonder if we also use it to initialize odp? What do we gain by auto init of odp_manager before the datafile is loaded if we drop all calls beforehand anyway (at least in py/rb)? This would allow us to avoid unnecessary allocation.

Reply:
@zashraf1985 I see it's a question about enabling or disabling ODP by default. Some clients won't use ODP, so this ODP initialization by default will be extra cost for them.

I see enabling it by default will help for clients who will use ODP and hope they'll be major soon.
Instead, for existing clients who do not want to use ODP and avoid extra overhead, we let them "disable" odp integration, which will be bypass all instantiation/initialization. If they do not care much, they can let it initialized by default, but all odp tasks will be ignored because odp not integrated in datafile.